js/tests/friend-support_test.mjs

```diff
@@ -0,0 +1,125 @@
+import { once } from 'node:events'
+import * as cp from 'node:child_process'
+import { mkdir, writeFile, chmod } from 'fs/promises'
+import { tmpdir } from 'node:os'
+import { join } from 'path'
+
+export const tests = []
+const fetch = _fetch // to redefine the real fetch
+
+const port = 5000
+
+export const setup = async ({ randStr }) => {
+  const dir = '.'
+
+  await mkdir(`${dir}/guests`, { recursive: true })
+
+  const randLastName = randStr()
+
+  const createFilesIn = ({ files, dirPath }) =>
+    Promise.all(
+      files.map(([fileName, content]) =>
+        writeFile(`${dirPath}/${fileName}`, JSON.stringify(content)),
+      ),
+    )
+
+  const sendRequest = async (path, options) => {
+    const response = await fetch(`http://localhost:${port}${path}`, options)
+    const { status, statusText, ok } = response
+    const headers = Object.fromEntries(response.headers)
+    let body = ''
+    try {
+      body = await response.json()
+    } catch (err) {
+      body = err
+    }
+    return { status, body, headers, randLastName }
+  }
+
+  return { tmpPath: dir, createFilesIn, sendRequest }
+}
+
+tests.push(async ({ path, ctx }) => {
```

Comment:
can ctx be further destructured? `ctx: { server }` at least?

Reply:
The server is now spawned in every test, therefore I kept `ctx` instead of expanding it to the function used from this object